awscli/customizations/s3/results.py

```diff
@@ -246,8 +246,8 @@ def _get_ongoing_dict_key(self, result):
                 'Any result using _get_ongoing_dict_key must subclass from '
                 'BaseResult. Provided result is of type: %s' % type(result)
             )
-        return ':'.join(
-            str(el) for el in [result.transfer_type, result.src, result.dest])
+        properties = [result.transfer_type, result.src, result.dest]
+        return u':'.join(six.text_type(p) for p in properties)
```

Comment:
Do we know if these are guaranteed to already be decoded and so they'll always be either ASCII only ``str`` *or* ``unicode`` objects? Because blindly casting to ``unicode`` on 2.7 will raise an error if the ``str`` contains bytes that are not ASCII encoded. e.g.:

```pycon
>>> unicode(u"☃".encode("utf8"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 0: ordinal not in range(128)
```

If we know that isn't going to be an issue, then we're good and we can just merge this. If not, we might have to do something more here.